packages/nextjs-kit/src/types/index.ts

```diff
@@ -0,0 +1,15 @@
+export interface HeaderData {
+  title: string;
+  href: string;
+  align?: string;
+}
+
+export interface FooterProps {
+  linkText: string;
+  href: string;
+}
+type Parent = 'parent' | 'parentId';
+
+export type FooterMenuItem = FooterProps & {
+  [key in Partial<Parent>]?: string | null;
```

Comment:
I don't think the partial actually does what we thought here. The `?` should be enough to make it optional.
```suggestion
  [key in Parent]?: string | null;
```